services/venice-server/src/main/java/com/linkedin/venice/stats/ServerHttpRequestStats.java

```diff
@@ -445,11 +445,15 @@ public void recordEarlyTerminatedEarlyRequest() {
   }
 
   public void recordKeySizeInByte(int keySize) {
-    requestKeySizeSensor.record(keySize);
+    if (requestKeySizeSensor != null) {
+      requestKeySizeSensor.record(keySize);
+    }
```

Comment:
Hmm... but it's weird though... why is this sensor null?

The server config should control both whether this sensor is null, and also whether this function gets called or not 🤔

Reply:
What I mean is:

This function is only called from `MultiGetResponseStatsWithSizeProfiling` (and the equivalent compute-related class). And `MultiGetResponseStatsWithSizeProfiling` can only get constructed when `serverConfig.isKeyValueProfilingEnabled()` is true...

And `AggServerHttpRequestStats` also gets passed the same config, which (after a few indirections) should determine whether that sensor gets created or not.

So how can these be mismatched?

Reply:
The issue is that sensor creation is controlled by a configuration flag (`isKeyValueProfilingEnabled`), but the reporting logic is not. Hence reporting just blindly tries to update sensor even when `isKeyValueProfilingEnabled == false`

Reply:
I think this is coming from `SingleGetResponseStats::recordMetrics` --> `ResponseStatsUtil::recordValueSizeInByte`

Reply:
```
```

Reply:
This is regression due to: https://github.com/linkedin/venice/pull/1377

Reply:
Sorry about that. Any idea why this wasn't caught in any of the tests?

Reply:
Oh yeah, right. I see it now: https://github.com/linkedin/venice/pull/1377#discussion_r1878842022

Reply:
What's the motivation to disable KV size profiling for single gets?

Reply:
Metric bloat maybe? IDK. @nisargthakkar ?

Reply:
>Any idea why this wasn't caught in any of the tests?

Probably we don't have E2E a test which verifies all metrics for single-get request type?